utils/userStatus.js

```diff
@@ -302,6 +303,150 @@ const getNextDayTimeStamp = (timeStamp) => {
   return nextDateDateTime.getTime();
 };
 
+/**
+ * Generates discord nickname for a user
+ *
+ * @param {string} username - The discord username of the user.
+ * @returns {string} - Nickname of the user.
+ */
+const generateOOONickname = (username, from, until) => {
+  if (!from && !until) return username;
+  const untilDate = new Date(Number(until));
+  const untilDay = untilDate.getDate();
+  const untilMonth = month[untilDate.getMonth()];
+
+  let oooMessage;
+
+  if (from && until) {
+    const fromDate = new Date(Number(from));
+    const fromDay = fromDate.getDate();
+    const fromMonth = month[fromDate.getMonth()];
+
+    oooMessage = `(OOO ${fromMonth} ${fromDay} - ${untilMonth} ${untilDay})`;
+  } else {
+    oooMessage = `(OOO ${untilMonth} ${untilDay})`;
+  }
+
+  // the max length of the nickname should be discord nickname length limit - OOO message length
+  // the extra 1 is for the space between ooo date and the nickname
+  const nicknameLen = discordNicknameLength - oooMessage.length - 1;
+  return `${username.substring(0, nicknameLen)} ${oooMessage}`;
+};
+
+/**
+ * @param userId { string }: Id of the User
+ * @param status { object: { from?: number, until: number }}: OOO date object
+ * @returns Promise<object>
+ */
+const updateNickname = async (userId, status = {}) => {
+  try {
+    const { discordId, username } = await userServices.getUserDiscordIdUsername(userId);
+    try {
+      const nickname =
+        status.from || status.until ? generateOOONickname(username, status.from, status.until) : username;
+
+      await userStatusServices.updateDiscordUserNickname(discordId, nickname);
+    } catch (err) {
+      logger.error("Failed to update user's nickname");
+    }
+  } catch (err) {
+    logger.error(`Failed to get discord id and username for user with id ${userId}`);
+    throw err;
+  }
+};
+
+const updateUsersDiscordNicknameBasedOnStatus = async (usersNicknameUpdates) => {
+  let errorsLen = 0;
+  try {
+    const updates = await Promise.allSettled(usersNicknameUpdates);
+    for (const update of updates) {
+      if (update.status === "rejected") {
+        errorsLen++;
+      }
+    }
+
+    if (errorsLen) {
```

Comment:
we are only sending the error metric. what about the success metric i.e how many user names were updated successfully.